Userland/Libraries/LibVideo/PlaybackManager.cpp

```diff
@@ -509,20 +561,32 @@ class PlaybackManager::PausedStateHandler : public PlaybackManager::PlaybackStat
     PlaybackState get_state() const override { return PlaybackState::Paused; }
 };
 
+// Wait for the buffer to fill as if it decodes at 60fps.
+// FIXME: Make this smarter, perhaps time decodes in the decoding thread and use two atomics to track
+//        the average, or find a simple way to send an event from the decoding thread to trigger a resume.
+constexpr int frame_decoding_time = 16;
+
 class PlaybackManager::BufferingStateHandler : public PlaybackManager::ResumingStateHandler {
     using PlaybackManager::ResumingStateHandler::ResumingStateHandler;
 
     ErrorOr<void> on_enter() override
     {
-        manager().m_decode_timer->start(0);
+        manager().m_present_timer->start(frame_decoding_time * frame_buffer_count);
         return {};
     }
 
     StringView name() override { return "Buffering"sv; }
 
-    ErrorOr<void> on_buffer_filled() override
+    ErrorOr<void> on_timer_callback() override
     {
-        return assume_next_state();
+        auto buffer_is_full = manager().m_buffer_is_full.load();
+        dbgln_if(PLAYBACK_MANAGER_DEBUG, "Buffering timer callback has been called. Buffer is {}.", buffer_is_full ? "full, exiting"sv : "not full, waiting"sv);
+        if (buffer_is_full) {
+            return assume_next_state();
+        }
+
+        manager().m_present_timer->start(frame_decoding_time);
```

Comment:
Maybe create an accessor method `present_timer()` instead of accessing the member directly?

Reply:
This pattern is present in the rest of the code as well, so I guess it's fine!